### PR TITLE
prepare package for cnpm

### PR DESCRIPTION
**Pull request by chak3r**

### CODE REVIEW

File: cmake/build_amd64.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+file(
+    TO_CMAKE_PATH
+    "$ENV{ENV_INSTALL}"
+    ROOT
+)
+
+set(
+    BUILD
+    3
+)
```
Comment (gshep) [inline cmake/build_amd64.cmake]: maybe it's better to use env var

File: package.cmake [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		PROPERTIES
+		IMPORTED_LOCATION
+		    "${CMAKE_CURRENT_LIST_DIR}/lib/gtest_main.a"
+		IMPORTED_LOCATION_DEBUG
+		    "${CMAKE_CURRENT_LIST_DIR}/lib/gtestd_maind.a"
+		INTERFACE_INCLUDE_DIRECTORIES
+		    "${CMAKE_CURRENT_LIST_DIR}/include/gtest"
+	    )
+	endif()
+
```
Comment (gshep) [inline package.cmake:55]: empty line

File: package.cmake [hunk 3]
```diff
@@ -0,0 +1,77 @@
+if(NOT TARGET GTest::GTest)
+	add_library(
+		GTest::GTest
+		STATIC
+		IMPORTED
+	)
+
+        if(TARGET Threads::Threads)
+            set_target_properties(GTest::GTest PROPERTIES
+                INTERFACE_LINK_LIBRARIES Threads::Threads)
```
Comment (gshep) [inline package.cmake]: make conformant to the rest

File: googletest/CMakeLists.txt [hunk 4]
```diff
@@ -284,3 +284,31 @@ if (gtest_build_tests)
...
+  PROPERTIES
+    PREFIX
+    ""
+)
+
+set_target_properties(
+  gtest_main
+  PROPERTIES
+    PREFIX
+    ""
```
Comment (gshep) [inline googletest/CMakeLists.txt]: unite to one call

**Review by gshep — CHANGES_REQUESTED** [review]

File: cmake/build_amd64.cmake (hunk 1 above)
Comment (chak3r) [reply cmake/build_amd64.cmake]: fixed

File: package.cmake (hunk 2 above)
Comment (chak3r) [reply package.cmake:55]: fixed?

File: package.cmake (hunk 3 above)
Comment (chak3r) [reply package.cmake]: fixed

File: googletest/CMakeLists.txt (hunk 4 above)
Comment (chak3r) [reply googletest/CMakeLists.txt]: done

File: cmake/build_common.cmake [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+)
+
+set(
+    BUILD_PATH
+    "${CMAKE_CURRENT_LIST_DIR}/../build"
+)
+
+set(
+    PACKAGE_NAME
+    "gtest-${VERSION}-${ARCH}-${BUILD}$ENV{TAG}"
```
Comment (gshep) [inline cmake/build_common.cmake]: just TAG, non ENV{TAG}

**Review by gshep — CHANGES_REQUESTED** [review]

File: cmake/build_common.cmake (hunk 5 above)
Comment (chak3r) [reply cmake/build_common.cmake]: fixed

**Review by gshep — APPROVED** [review]

**Pull request merged**